src/sentry_value.c

```diff
@@ -624,6 +666,18 @@ sentry__value_as_uuid(sentry_value_t value)
 char *
 sentry__value_stringify(sentry_value_t value)
 {
+#define STRINGIFY_NUMERIC(fmt, value_fn)                                       \
+    do {                                                                       \
+        char buf[24];                                                          \
+        size_t written                                                         \
+            = (size_t)sentry__snprintf_c(buf, sizeof(buf), fmt, value_fn);     \
+        if (written >= sizeof(buf)) {                                          \
+            return sentry__string_clone("");                                   \
+        }                                                                      \
+        buf[written] = '\0';                                                   \
```

Comment:
`STRINGIFY_NUMERIC` could use a comment (above the `#define`) that signals that we willingly conflate the two error cases of `snprintf` (return value is an `int` because written can be < 0, which signifies an encoding error) into an overflowing `size_t`, because we handle them the same way (return an empty string).